src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,82 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+import { Filter } from '../../types/FIlter';
+
+type Props = {
+  filter: Filter
+  updateFilter: (newFilter: Filter) => void,
+  updateQuery: (newQuery: string) => void
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filter,
+  updateFilter,
+  updateQuery,
+}) => {
+  const [valueSelect, setValueSelect] = useState<Filter>(filter);
+  const [valueSearch, setValueSearch] = useState('');
```

Comment:
You don't need to create additional states in this component. You already have it inside the App
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/99403198/6017a08e-62df-4dea-a41d-059464acd2c1)